src/main/java/com/denizenscript/clientizen/scripts/containers/gui/elements/BarElement.java

```diff
@@ -0,0 +1,100 @@
+package com.denizenscript.clientizen.scripts.containers.gui.elements;
+
+import com.denizenscript.clientizen.scripts.containers.gui.GuiScriptContainer;
+import com.denizenscript.denizencore.tags.TagContext;
+import com.denizenscript.denizencore.utilities.YamlConfiguration;
+import com.denizenscript.denizencore.utilities.debugging.Debug;
+import io.github.cottonmc.cotton.gui.widget.WBar;
+import io.github.cottonmc.cotton.gui.widget.WWidget;
+import io.github.cottonmc.cotton.gui.widget.data.Texture;
+import net.minecraft.screen.PropertyDelegate;
+import net.minecraft.text.Text;
+
+import static com.denizenscript.clientizen.scripts.containers.gui.GuiScriptContainer.getTaggedEnum;
+import static com.denizenscript.clientizen.scripts.containers.gui.GuiScriptContainer.getTaggedInt;
+import static com.denizenscript.clientizen.scripts.containers.gui.GuiScriptContainer.getTaggedString;
+import static com.denizenscript.clientizen.scripts.containers.gui.GuiScriptContainer.parseTexture;
+
+public class BarElement implements GuiScriptContainer.GuiElementParser {
+
+    // <--[language]
+    // @name Bar GUI Element
+    // @group GUI System
+    // @description
+    // Bars are GUI elements that work as a progress bar; they have a UI type of "bar".
```

Comment:
I suggest this be called `ProgressBarElement`/`progress_bar`

Reply:
Renamed